### PR TITLE
do not render grouped view if groupable attribute is false

### PR DESCRIPTION
**Pull request by msh-odoo**
PURPOSE
Do not render grouped view if view is not groupable

SPEC
View should be rendered ungrouped if view is not groupable even if default group by is passed.

TASK 1969134

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by dbo-odoo** [comment]
@aab-odoo 
This is more for you than for me...
I'm honestly not sure how this should be done after the controlpanelview refactor though.

How to reproduce:
- Go to Documents Kanban view
- Enable Studio
- Set default group by to some field
- Close Studio => Crash, because the view is not groupable but it tries to group itself anyway

**Comment by dbo-odoo** [comment]
@aab-odoo 
This is more for you than for me...
I'm honestly not sure how this should be done after the controlpanelview refactor though.

How to reproduce:
- Go to Documents Kanban view
- Enable Studio
- Set default group by to some field
- Close Studio => Crash, because the view is not groupable but it tries to group itself anyway

**Comment by dbo-odoo** [comment]
@aab-odoo 
This is more for you than for me...
I'm honestly not sure how this should be done after the controlpanelview refactor for higher versions though.

How to reproduce:
- Go to Documents Kanban view
- Enable Studio
- Set default group by to some field
- Close Studio => Crash, because the view is not groupable but it tries to group itself anyway

File: addons/web/static/src/js/views/abstract_view.js [hunk 1]
```diff
@@ -162,9 +162,12 @@ var AbstractView = Class.extend({
             this.controllerParams.withControlPanel = params.withControlPanel;
         }
 
-        var groupBy = params.groupBy;
-        if (typeof groupBy === 'string') {
-            groupBy = [groupBy];
+        // set groupBy only if view is groupable
+        if (this.groupable) {
+            var groupBy = params.groupBy;
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/abstract_view.js]: define groupby outside the if (and initialize it to [])

File: addons/web/static/src/js/views/kanban/kanban_view.js [hunk 2]
```diff
@@ -36,7 +36,9 @@ var KanbanView = BasicView.extend({
         // false so that they will won't be loaded by the initial load
         this.loadParams.openGroupByDefault = config.device.isMobile ? false : true;
         this.loadParams.type = 'list';
-        this.loadParams.groupBy = this.arch.attrs.default_group_by ? [this.arch.attrs.default_group_by] : (params.groupBy || []);
+        if (this.groupable) {
+            this.loadParams.groupBy = this.arch.attrs.default_group_by ? [this.arch.attrs.default_group_by] : (params.groupBy || []);
+        }
```
Comment (aab-odoo) [inline addons/web/static/src/js/views/kanban/kanban_view.js]:
```
} else {
    this.loadParams.groupBy = [];
}

File: addons/web/static/tests/views/abstract_view_tests.js [hunk 3]
```diff
@@ -130,5 +131,42 @@ QUnit.module('Views', {
...
+                    group_by: ['bar'],
+                },
+            }],
+            archs: {
+                'foo,false,list': '<tree><field name="foo"/><field name="bar"/></tree>',
+                'foo,false,search': '<search></search>',
+            },
+            data: this.data,
+            mockRPC: function (route, args) {
+                if (args.method === 'read_group' && args.kwargs.groupby[0] === "bar") {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/abstract_view_tests.js]: I'd remove the second part of the condition

File: addons/web/static/tests/views/kanban_tests.js [hunk 4]
```diff
@@ -2616,6 +2616,38 @@ QUnit.module('Views', {
...
+            View: KanbanView,
+            model: 'partner',
+            data: this.data,
+            arch: '<kanban class="o_kanban_test" default_group_by="bar">' +
+                '<field name="bar"/>' +
+                '<templates><t t-name="kanban-box">' +
+                '<div><field name="foo"/></div>' +
+                '</t></templates></kanban>',
+            mockRPC: function (route, args) {
+                if (route === '/web/dataset/call_kw/partner/read_group') {
```
Comment (aab-odoo) [inline addons/web/static/tests/views/kanban_tests.js]: args.method === 'read_group'?

**Comment by aab-odoo** [comment]
@dbo-odoo It's looking good. Do you want me to r+?

**Comment by dbo-odoo** [comment]
@robodoo r+

@aab-odoo i mostly concerned about the forward port given the changes with the controlpanelview; i won't be able to provide guidance for that...

**Comment by aab-odoo** [comment]
@dbo-odoo The CP branch hasn't been merged yet ^^

**Pull request closed without merging**